### PR TITLE
feat(ctb): FDG bond safety mode

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds a safety mode to the `FaultDisputeGame` that can be triggered by the `SuperchainConfig`'s guardian role. This mode will return all bonds back to their submitters when `claimCredit` is called. This is intended to mitigate situations where the honest challenger is playing a very high-stakes dispute game, and there is risk of losing a massive amount of money due to a bug in the system.

### Semantic Changes

- A new function, `enableSafetyMode`, has been added. This function may only be called by the `Guardian`.
  - This function toggles the `safetyMode` flag.
- When `claimCredit` is called:
  - If `safetyMode` is false, the `_recipient`'s `credit` is distributed to them.
  - If `safetyMode` is true, the `_recipient`'s `bonds` are returned to them.

### Open Questions

- The `Guardian` role is slow-moving, and requires multisig signers to gather. Do we want a new role for this action?
- What should the `BOND_PAYOUT_DELAY` be?
- The safety mode flag is currently per-game. This should likely live in another contract so that it may be set globally?

### TODO
-  [ ] Have the `op-challenger` wait the `BOND_PAYOUT_DELAY` after a game has resolved to claim their bonds.

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* **#9591** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9591?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9591?utm_source=stack-comment).

File: packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol [hunk 1]
```diff
@@ -68,6 +75,10 @@
     /// @notice Credited balances for winning participants.
     mapping(address => uint256) public credit;
 
+    /// @notice An internal mapping of claimants to the amounts of bonds they have placed. This is used in safety
+    ///         mode, where the bond is returned to the claimants upon resolution.
+    mapping(address => uint256) public bonds;
```
[the remark this answers is not in the document]
Comment (clabby) [reply packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol:80]: Reentrancy test added

**Review by Inphi — COMMENTED** [review]
What's the status of this PR? Are we still using this approach for safety mode?

**Comment by clabby** [comment]
> What's the status of this PR? Are we still using this approach for safety mode?

Nah, don't think so. Will move into draft for reference in case the other plan doesn't work out.

**Pull request closed without merging**